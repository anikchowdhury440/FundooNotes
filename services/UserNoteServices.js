import Firebase from '../config/Firebase'

class UserNoteServices {
    storeNoteinDatabase = (userid, noteKey, notes) => {
        return new Promise((resolve, reject) => {
            Firebase.database().ref('UserNotes/' + userid + '/' + noteKey).set({
                notes : notes
            })
            .then(() => resolve('success')) 
            .catch((error) => reject(error))  
        }) 
    }

    getNoteFromDatabase = (userid) => {
        return new Promise((resolve, reject) => {
            Firebase.database().ref('UserNotes/' + userid).once('value')
                .then(snapshot => resolve(snapshot.val()))
                .catch(error => reject(error))
        })
    }

    updateNoteInFirebase = (userid, noteKey, notes) => {
        return new Promise((resolve, reject) => {
            Firebase.database().ref('UserNotes/' + userid  + '/' + noteKey).set({
                notes : notes
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    deleteNoteInFirebase = (userid, notekey, notes) => {
        return new Promise((resolve, reject) => {
            Firebase.database().ref('UserNotes/' + userid  + '/' + notekey).set({
                notes : notes
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    restoreNoteInFirebase = (userid, notekey) => {
        return new Promise((resolve, reject) => {
            Firebase.database().ref('UserNotes/' + userid  + '/' + notekey + '/notes').update({
                isDeleted : 0
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    removeNoteInFirebase = (userid, notekey) => {
        return new Promise((resolve, reject) => {
            Firebase.database().ref('UserNotes/' + userid  + '/' + notekey).remove()
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    updateNoteLabelInFirebase = (userid, notekey, labelId) => {
        return new Promise((resolve, reject) => {
            Firebase.database().ref('UserNotes/' + userid  + '/' + notekey + '/' + 'notes').update({
                labelId : labelId
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }

    storeNoteinDatabaseAPI = (userid, noteKey, notes) => {
        return new Promise((resolve, reject) => {
            fetch(`https://fundoonotes-893f7-default-rtdb.firebaseio.com/UserNotes/${userid}/${noteKey}.json`,{
                method: 'PUT',
                headers: {
                  'Content-Type': 'application/json'
                },
                body: JSON.stringify({
                   notes : notes
                })
            })
            .then(() => resolve('success')) 
            .catch((error) => reject(error))  
        }) 
    }

    removeNoteinDatabaseAPI = (userid, notekey) => {
        return new Promise((resolve, reject) => {
            fetch(`https://fundoonotes-893f7-default-rtdb.firebaseio.com/UserNotes/${userid}/${notekey}.json`,{
                method: 'DELETE'
            })
            .then(() => resolve('success'))
            .catch(error => reject(error))
        })
    }
}

export default new UserNoteServices();