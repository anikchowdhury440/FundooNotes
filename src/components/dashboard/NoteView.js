import React, {Component} from 'react';
import {ScrollView, View} from 'react-native';
import {Button, Card, Title, Paragraph} from 'react-native-paper';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { Strings } from '../../Language/Strings';
import * as Keychain from 'react-native-keychain';
import NoteViewStyle from '../../styles/NoteView.style';
import UserNoteServices from '../../../services/UserNoteServices';

export default class MainView extends Component {
    constructor(props) {
        super(props);
        this.state = {
            userNotes : []
        }
    }

    async componentDidMount() {
        const credential = await Keychain.getGenericPassword();
        const UserCredential = JSON.parse(credential.password);
        UserNoteServices.getNoteFromDatabase(UserCredential.user.uid)
            .then(async data => {
                let notes = data ? data : {}
                await this.setState({
                    userNotes : notes
                })
            })
    }

    selectNote = (noteKey) => {
        this.props.navigation.push('AddNote', { noteKey : noteKey, notes : this.state.userNotes[noteKey].notes})
    }

    render() {
        let noteID = Object.keys(this.state.userNotes);
        return (
            <ScrollView style = {{marginBottom : 60}}>
                <View style = {{flexDirection: 'row', flexWrap: 'wrap'}}>
                    { noteID.length > 0 ?
                        noteID.reverse().map(key => (
                            <Card
                                key = {key}
                                style = {(this.props.listView) ? NoteViewStyle.list_item_style :  NoteViewStyle.list_item_grid_style }
                                onPress = {() => this.selectNote(key)} >
                                <Card.Content>
                                    <Title 
                                        style = {(this.state.userNotes[key].notes.note == '') ? NoteViewStyle.list_title_note_empty_style : NoteViewStyle.list_title_style}>
                                            {this.state.userNotes[key].notes.title}
                                    </Title>
                                    <Paragraph
                                        style = {(this.state.userNotes[key].notes.title == '') ? NoteViewStyle.list_note_title_empty_style : NoteViewStyle.note_description_style}>
                                            {this.state.userNotes[key].notes.note}
                                    </Paragraph>
                                </Card.Content>  
                            </Card>
                        )) 
                        :
                        null
                    }
                </View>
                <Button 
                    onPress = { async () => {
                        await AsyncStorage.setItem('isLoggedIn', JSON.stringify(false));
                        this.props.navigation.push('Login')
                    }}>
                        {Strings.logout}
                    </Button>      
            </ScrollView>
        )
    }
}