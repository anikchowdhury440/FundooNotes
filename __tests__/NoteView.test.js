import React from 'react';
import {shallow} from 'enzyme';
import NoteView from '../src/components/dashboard/NoteView';
import UserNoteServices from '../services/UserNoteServices';

describe('test NoteView', () => {
    it('should match to snapshot', () => {
        const component = shallow(<NoteView />)
        expect(component).toMatchSnapshot();
    })

})