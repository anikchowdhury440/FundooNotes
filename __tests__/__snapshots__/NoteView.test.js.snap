// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test NoteView should match to snapshot 1`] = `ShallowWrapper {}`;
